apps/condo/domains/miniapp/access/B2BAppContext.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * Generated by `createschema miniapp.B2BAppContext 'integration:Relationship:B2BApp:PROTECT; organization:Relationship:Organization:PROTECT; settings:Json; state:Json;'`
+ */
+
+const get = require('lodash/get')
+const { getById } = require('@core/keystone/schema')
+const { throwAuthenticationError } = require('@condo/domains/common/utils/apolloErrorFormatter')
+const { checkOrganizationPermission } = require('@condo/domains/organization/utils/accessSchema')
+const { checkB2BAppAccessRight } = require('@condo/domains/miniapp/utils/accessSchema')
+const { SERVICE } = require('@condo/domains/user/constants/common')
+
+
+/**
+ * B2B App context may only be read:
+ * 1. Admin / support
+ * 2. Organization employee (without settings / state)
+ * 3. Integration service account
+ */
+async function canReadB2BAppContexts ({ authentication: { item: user } }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+
+    if (user.isAdmin || user.isSupport) return {}
+
+
+    return {
+        OR: [
+            { organization: { employees_some: { user: { id: user.id }, isBlocked: false, isRejected: false, deletedAt: null } } },
+            { app: { accessRights_some: { user: { id: user.id }, deletedAt: null } } },
+        ],
+    }
+}
+
+/**
+ * B2B App context may only be created by:
+ * 1. Admin / support
+ * 2. Organization employee with canManageIntegrations = true
+ *
+ * B2B App context may only be updated by:
+ * 1. Admin / support
+ * 2. App service user
+ */
+async function canManageB2BAppContexts ({ authentication: { item: user }, originalInput, operation, itemId }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+    if (user.isAdmin || user.isSupport) return true
+
+    if (operation === 'create') {
+        const organizationId = get(originalInput, ['organization', 'connect', 'id'])
+
+        return await checkOrganizationPermission(user.id, organizationId, 'canManageIntegrations')
+    } else if (operation === 'update') {
+        if (!itemId) return false
+        const context = await getById('B2BAppContext', itemId)
+        if (!context || context.deletedAt) return false
+        const { app } = context
+
+        return await checkB2BAppAccessRight(user.id, app)
+    }
+
+    return false
+}
+
+/**
+ * Organization employees not allowed seeing app's settings / state
+ */
+async function canReadAndManageSensitiveContextData ({ authentication: { item: user } }) {
+    return !!(user.isSupport || user.isAdmin || user.type === SERVICE)
```

Comment:
weak permission check

Reply:
We really check the service user type in case of billing mini app?

Reply:
It's not weak....it's a field permission and executed on top of list permission....

And this way allows to resolve all "usage cases" for field without making any subqueries

Reply:
> We really check the service user type in billing/acquiring mini app cases?

Not for now, but I guess we should